docs/style-guide.md

```diff
@@ -0,0 +1,201 @@
+
+```javascript
+new Intl.DisplayNames(undefined, {
+    type: "dateTimeField"
+}).of("timeZoneName");
+```
+
+##### Decision
+
+- The ergonomics of using the same identifier in multiple contexts outweights the desire to match the W3C convention.
+- We believe that most users will typically copy these strings off MDN, rather than typing them manually, weakening the importance of following programmer intuition.
+
+#### Alternative: Kebab case for new identifiers only
+
+A way to be closer to the W3C recommendation would be to use kebab case except when there's precedent for camel case in the identifier.  So, for `"timeZoneName"`, we would use camel case, but for a new identifier like `"except-zero"`, we would use kebab case.  Example:
+
+```javascript
+new Intl.DisplayNames(undefined, {
+    type: "date-time-field"
+}).of("timeZoneName");
+
+number.toLocaleString(undefined, {
+    signDisplay: "except-zero"
+});
+```
+
+##### Pros
+
+- Identifiers that are used in multiple contexts are consistent in the API across input, output, and keys.
+- Most identifiers follow programmer intuition; those that don't are the exception rather than the rule.
+
+##### Cons
+
+- Could introduce problems if a string is introduced as kebab case now and then in the future wants to be used as the key of an object literal.
+- Can introduce inconsistencies even in the same line of code, as shown in the example above.
+
+##### Decision
+
+- This hybrid approach creates more problems while not fully addressing the concerns of the kebab-case-only approach.
+
+#### Alternative: Use kebab case but also accept camel case
+
+ECMA-402 could allow users to choose between kebab case and camel case for relevant identifiers.  For example, when calling `Intl.DisplayNames.prototype.of`, the identifier could be either `"timeZoneName"` or `"time-zone-name"` and both would be treated equally.
+
+##### Pros
+
+- When writing code, programmers can follow their intuition.
+- We still allow programmers to use the same identifiers in multiple contexts.
+
+##### Cons
+
+- Output strings (those generated by library functions rather than written in user code) cannot have multiple cases; this alternative does not clarify how to handle such strings.
+- Could sow confusion as a result of multiple ways to achieve the same result.
+
+##### Decision
+
+- It is difficult to take this option seriously since it would add debt to the specification.
+- If strongly desired, the spec could be changed later to allow the kebab-cased strings.
+
+### Identifiers defined outside ECMA-402
+
+:star2: *ECMA-402 sits between ECMA-262 and internationalization standards such as UTS #35. For identifiers whose syntax comes from a different specification, follow the convention in that specification.* :star2:
+
+#### Examples
+
+Locale identifiers are defined in [UTS #35 Section 3](https://www.unicode.org/reports/tr35/#Identifiers).  Examples:
```

Comment:
I'd prefer something such as 
> [BCP47](http://www.rfc-editor.org/rfc/bcp/bcp47.txt) identifiers, using registered extensions as described in [UTS#35§3](https://www.unicode.org/reports/tr35/#Identifiers)

The identifiers are and must be valid BCP47.

Reply:
Done.